lib/semian/circuit_breaker.rb

```diff
@@ -30,9 +32,7 @@ def acquire
     end
 
     def request_allowed?
-      return true if closed?
-      half_open if error_timeout_expired?
-      !open?
+      closed? || half_open? || (open? && error_timeout_expired?)
```

Comment:
Might want to leave a comment that `open? && error_timeout_expired?` is intended as a non-mutating equivalent check to "should transition to half-open"?

Or should that condition even just go in `half_open?` to begin with?